src/credential-status/status-token.ts

```diff
@@ -0,0 +1,159 @@
+import AbortController from 'abort-controller'
+import fetch from 'node-fetch'
+import { cborDecode, cborEncode } from '../cbor'
+import { Tag } from '../cbor/cbor-x'
+import type { CoseKey } from '../cose'
+import { CoseType, CoseTypeToTag, Mac0, Sign1 } from '../cose'
+import { CWT } from '../cwt'
+import type { StatusArray } from './status-array'
+import { StatusList } from './status-list'
+
+interface CWTStatusTokenOptions {
+  claimsSet: {
+    statusArray: StatusArray
+    aggregationUri?: string
+    expirationTime?: number
+    timeToLive?: number
+  }
+  type: CoseType
+  key: CoseKey
+}
+
+interface CWTStatusTokenVerifyOptions {
+  token: Uint8Array
+  key?: CoseKey
+}
+
+interface CWTStatusVerifyOptions extends CWTStatusTokenVerifyOptions {
+  index: number
+  expectedStatus: number
+}
+
+enum CwtProtectedHeaders {
+  TYPE = 16,
+}
+
+enum CwtStatusListClaims {
+  StatusListUri = 2,
+  ExpirationTime = 4,
+  IssuedAt = 6,
+  StatusList = 65533,
+  TimeToLive = 65534,
+}
+
+export class CWTStatusToken {
+  static async build(options: CWTStatusTokenOptions): Promise<Uint8Array> {
+    const cwt = new CWT()
+    cwt.setHeaders({
+      protected: {
+        [CwtProtectedHeaders.TYPE]: 'application/statuslist+cwt',
```

Comment:
We usually write enum keys as PascalCase

```suggestion
        [CwtProtectedHeaders.Type]: 'application/statuslist+cwt',
```